primitives/src/lib.rs

```diff
@@ -13,14 +17,227 @@ use sp_std::{
     prelude::*,
 };
 
+pub use bitcoin::types::H256Le;
+
+pub mod issue {
+    use super::*;
+
+    #[derive(Encode, Decode, Clone, PartialEq)]
+    #[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+    pub enum IssueRequestStatus {
+        Pending,
+        /// optional refund ID
+        Completed(Option<H256>),
+        Cancelled,
+    }
+
+    impl Default for IssueRequestStatus {
+        fn default() -> Self {
+            IssueRequestStatus::Pending
+        }
+    }
+
+    // Due to a known bug in serde we need to specify how u128 is (de)serialized.
+    // See https://github.com/paritytech/substrate/issues/4641
+    #[derive(Encode, Decode, Default, Clone, PartialEq)]
+    #[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+    pub struct IssueRequest<AccountId, BlockNumber, Wrapped, Collateral> {
+        pub vault: AccountId,
+        pub opentime: BlockNumber,
+        pub period: BlockNumber,
```

Comment:
Would be great to have doc comments on all public fields, but this is obviously not a blocker.